src/presets/vercel/utils.ts

```diff
@@ -10,13 +10,31 @@ import type {
   VercelServerlessFunctionConfig,
 } from "./types";
 
+const MIN_NODE_VERSION = 16;
+const MAX_NODE_VERSION = 20;
+
+function getSystemNodeVersion() {
+  const systemNodeVersion = Number.parseInt(
+    process.versions.node.split(".")[0]
+  );
+
+  return Number.isNaN(systemNodeVersion) ? 20 : systemNodeVersion;
+}
+
 export async function generateFunctionFiles(nitro: Nitro) {
   const buildConfigPath = resolve(nitro.options.output.dir, "config.json");
   const buildConfig = generateBuildConfig(nitro);
   await writeFile(buildConfigPath, JSON.stringify(buildConfig, null, 2));
 
-  const systemNodeVersion = process.versions.node.split(".")[0];
-  const runtimeVersion = `nodejs${systemNodeVersion}.x`;
+  const systemNodeVersion = getSystemNodeVersion();
+  const closestLtsVersion =
+    systemNodeVersion % 2 === 0 ? systemNodeVersion : systemNodeVersion - 1;
```

Comment:
I think instead we can use an array of supported Node.js versions in vercel and clamp (up) to closest

Reply:
Are you thinking something like this? 

```ts
  let usedNodeVersion = 0;
  for (const version of SUPPORTED_NODE_VERSIONS) {
    if (Math.abs(systemNodeVersion - version) <= Math.abs(systemNodeVersion - usedNodeVersion)) {
      usedNodeVersion = version
    }
  }
```

Reply:
Something like this could be nice 👍🏼 I think if `SUPPORTED_NODE_VERSIONS` is sorted array, we can find first upper abound with simple find as well.